### PR TITLE
#162949030 Setup environment for writing tests

### PR DESCRIPTION
**Pull request by jomasim**
#### What does this PR do?
This PR sets up environment for writing tests.
#### Description of Task to be completed?
- [x] installed enzyme package
- [x] created setupTest.js file
- [x] updated app.test to use enzyme shallow instead of react-dom

#### How should this be manually tested?

1. Clone this repository on your local machine using 
```
 git clone https://github.com/andela/ah-robotics-frontend.git 
```
2. Change your working directory to the project's directory using 
```
cd ah-robotics-frontend
```
3. Change to the specific branch `ch-setting-testing-environment-162949030` using 
```
git checkout  ch-setting-testing-environment-162949030
```

4. Run `npm install` to install all the modules listed in the `package.json` file.

5. Run `npm test` command to run the tests.

6. Choose  **a**-**option** to run all tests
screenshot 2019-02-10 at 13 07 31

screenshot 2019-02-07 at 11 02 58

[#162949030](https://www.pivotaltracker.com/n/projects/2235171/stories/162949030)

### CODE REVIEW

**Review by sharkdevs — COMMENTED** [review]
There is a screen that comes up on running `npm test` which gives options to run the tests. Consider adding that description on the PR lest it takes a user by surprise. 
screen shot 2019-02-07 at 11 30 12

**Comment by jomasim** [comment]
> There is a screen that comes up on running `npm test` which gives options to run the tests. Consider adding that description on the PR lest it takes a user by surprise.
> screen shot 2019-02-07 at 11 30 12

I have added --watchAll flag to enable you run all tests by default

**Comment by Teatoller** [comment]
Hello @jomasim 
I have been able to test this feature branch and it is working as anticipated.
Regards
Steven Ennis

**Review by waracci — COMMENTED** [review]
Works as stipulated in your PR message!

screenshot 2019-02-07 at 12 57 56


Consider adding a blank line at the end of `setupTests.js` Github is complaining about it.

**Review by Paulvitalis200 — COMMENTED** [review]
Hey @jomasim,
I have been able to test this feature locally on my machine and it works as expected.

File: src/App.test.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
-import ReactDOM from 'react-dom';
+import { shallow } from 'enzyme';
 import App from './App';
 
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+describe('RootComponent', () => {
+  it('app should render without crashing ', () => {
```
Comment (Collins33) [inline src/App.test.js]: Remove the space after the crashing word

**Review by Collins33 — CHANGES_REQUESTED** [review]

File: src/App.test.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-import ReactDOM from 'react-dom';
+import { shallow } from 'enzyme';
 import App from './App';
 
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+describe('RootComponent', () => {
+  it('app should render without crashing', () => {
```
Comment (jomasim) [inline src/App.test.js:6]: @Collins33 I have removed the space.It looks fine now

**Review by Teatoller — CHANGES_REQUESTED** [review]
Hello Joma,

I have tested your feature branch and the noticeable difference is not that there is no test present.
You may need to revise the screenshot you have to the following.

screenshot 2019-02-10 at 12 24 19


Regards
Steven Ennis

**Comment by jomasim** [comment]
> Hello Joma,
> 
> I have tested your feature branch and the noticeable difference is not that there is no test present.
> You may need to revise the screenshot you have to the following.
> 
> screenshot 2019-02-10 at 12 24 19
> 
> Regards
> Steven Ennis

@Teatoller The test runner checks for any changes in the tests before it runs. We don't have any change right now so you need to press a to run the all the tests. Let me update the instructions to make it easy for you.

**Review by Teatoller — APPROVED** [review]

**Review by johnwayodi — COMMENTED** [review]
Nice work, I was able to test it locally and got the same output.

**Review by Collins33 — APPROVED** [review]

**Pull request merged**